File: app/src/main/java/com/eftimoff/bakingapp/recipelist/di/RecipeListComponent.java
package com.eftimoff.bakingapp.recipelist.di;

import com.eftimoff.bakingapp.app.injection.scopes.PerFragment;
import com.eftimoff.bakingapp.recipelist.view.RecipeListFragment;

import dagger.Subcomponent;

@PerFragment
@Subcomponent(modules = RecipeListModule.class)
public interface RecipeListComponent {

    void inject(RecipeListFragment recipeListFragment);

}

File: app/src/main/java/com/eftimoff/bakingapp/recipelist/di/RecipeListComponent.kt
package com.eftimoff.bakingapp.recipelist.di

import com.eftimoff.bakingapp.app.injection.PerFragment
import com.eftimoff.bakingapp.recipelist.view.RecipeListFragment
import dagger.Subcomponent

@PerFragment
@Subcomponent(modules = arrayOf(RecipeListModule::class))
interface RecipeListComponent {

    fun inject(recipeListFragment: RecipeListFragment)

}